test/pipe.js

```diff
@@ -4,20 +4,41 @@ var R = require('..');
 
 
 describe('pipe', function() {
-  function a(x) {return x + 'A';}
-  function b(x) {return x + 'B';}
-  function c(x) {return x + 'C';}
-  function d(x) {return x + 'D';}
 
-  it('executes its passed in functions in order from left to right', function() {
-    assert.strictEqual(R.pipe(a, b, c, d)(''), 'ABCD');
+  it('is a variadic function', function() {
+    assert.strictEqual(typeof R.pipe, 'function');
+    assert.strictEqual(R.pipe.length, 0);
   });
 
-  it('first function is passed multiple args', function() {
-    function e(a, b, c) {
-      return c + 'E';
-    }
-    assert.strictEqual(R.pipe(e, a, b, c)(1, 2, 3), '3EABC');
+  it('performs left-to-right function composition', function() {
+    var f = function(a) { return [a]; };
+    var g = function(a, b) { return [a, b]; };
+    var h = function(a, b, c) { return [a, b, c]; };
+
+    assert.strictEqual(R.pipe(f, f, f).length, 1);
+    assert.strictEqual(R.pipe(g, f, f).length, 2);
+    assert.strictEqual(R.pipe(h, f, f).length, 3);
+
+    assert.deepEqual(R.pipe(f, f, f)(1), [[[1]]]);
+    assert.deepEqual(R.pipe(g, f, f)(1, 2), [[[1, 2]]]);
+    assert.deepEqual(R.pipe(g, f, f)(1)(2), [[[1, 2]]]);
+    assert.deepEqual(R.pipe(h, f, f)(1, 2, 3), [[[1, 2, 3]]]);
+    assert.deepEqual(R.pipe(h, f, f)(1, 2)(3), [[[1, 2, 3]]]);
+    assert.deepEqual(R.pipe(h, f, f)(1)(2, 3), [[[1, 2, 3]]]);
+    assert.deepEqual(R.pipe(h, f, f)(1)(2)(3), [[[1, 2, 3]]]);
+
+    var $g = R.curry(g);
+
+    assert.strictEqual(R.pipe($g, $g).length, 2);
+    assert.deepEqual(R.pipe($g, $g)(1, 2)(3), [[1, 2], 3]);
+    assert.deepEqual(R.pipe($g, $g)(1)(2)(3), [[1, 2], 3]);
+  });
+
+  it('returns the identity function (the identity element of function composition) given no arguments', function() {
```

Comment:
Is there a use case for calling `pipe` with no arguments? It seems like this outcome might just be due to the implementation rather than something we want to explicitly test.